test/remake.jl

```diff
@@ -0,0 +1,118 @@
+using ModelingToolkit, SymbolicIndexingInterface
```

Comment:
Can you add JumpSystems to JumpProblems into this test to? (Which requires JumpProcesses?)

Note that JumpProblem is defined in JumpProcesses and has some remake related dispatches there that might need to be updated (basically they just forward all updates to the wrapped prob, i.e. `jump_problem.prob` which is usually a discrete or ODE problem).

Reply:
Resolved in https://github.com/SciML/SciMLBase.jl/pull/583/commits/d4d5466a90de0f586567e4d8aaa61c5518f9e4d1